docs/unification.md

```diff
@@ -0,0 +1,344 @@
+   equality between the two maps.
+2. Both maps contain no functions. In this case, apply AC unification to the
+   two maps.
+
+### AC Unification 
+
+First, we consider the following special cases:
+
+* If both maps are empty, discharge.
+* If one map has 0 elements and 1 variable, treat it as a variable pattern
+  instead, and perform variable abstraction on the other pattern, then follow
+the procedure for AC binding.
+* If both maps have 1 element and 0 variables, decompose with the keys and
+  values.
+* Otherwise, if one map is empty and the other has 0 elements, bind each
+  variable of the other map to the empty map.
+* Otherwise, if one map is empty (and the other has elements), fail.
+* Otherwise, perform variable abstraction on the elements of both maps and
+  decompose with the AC equation and free equations that result.
+
+### Variable Abstraction
+
+Variable abstraction replaces each element in one or both map patterns with
+a new fresh variable and generates a free equality equating that variable with
+the element it corresponds to. It then is left with an equality that consists
+purely of map concatenation and variables, and adds it to the AC equations to
+solve.
+
+### AC Binding
+
+If one term is a variable X and the other is `_Map_(s1, ..., sn)` where si is a
+variable, we consider the following cases:
+
+* If X is free, bind X to the other term.
+* If X is bound to a variable, bind the variable it is bound to to the other
+  term.
+* Otherwise, add the AC equation between the binding of X and the other term to
+  the AC equations to solve.
+
+### AC Subproblems
+
+After the priority queue is empty, we are left with one AC subproblem per map
+or set sort. We then solve each one according to the following algorithm, which
+returns a list of disjunct solutions:
+
+First, we replace each variable with its binding in the substitution until this
+can no longer occur.
+
+We are left with a system of equations P == s1 = s1' /\ ... /\ sp = sp' where
+aeach si and si' is of the form `_Map_(t1, ..., tk)` for some k > 1 where each
+ti is a variable. We define u1, ..., un to be the unique variables in the
+system of equations, and associate each ui an integer variable xi. We say a
+variable xi is *constrained* if it came from a variable abstraction of a map
+element, and *unconstrained* otherwise. We then convert P into the following
+system of linear Diophantine equations:
+
+d(u1, t1=t1')x1 + ... + d(un, t1=t1')xn = 0
+.
+.
+.
+d(u1, tp = tp')x1 + ... + d(un, tp=tp')xn = 0
+
+Where d(ui, ti = ti') is the number of occurrences of ui in ti minus the number
+of occurrences of ui in ti'.
+
+We then compute the set of positive, non-null, minimal solutions of this system
+of equations, called S, using an algorithm defined in
+"Competing for the AC-Unification Race" by Alexandre Boudet. Each solution Sj
+in S is one element of the *diophantine basis* of P. For each solution
```

Comment:
I mean, this sentence is defining that term. The diophantine basis of the problem P is the set of solutions to the diophantine equation system we generate for that problem. I agree the term isn't mentioned in the paper; it was a convenient term used in the Maude source code that I borrowed since it provided a useful way to refer to these individual elements that get combined to form the solutions to P. I imagine Steven Eker called it that because it's the solution to the system of diophantine equations, and it forms the basis (in the mathematical sense of the word basis) for the solutions to P.

Do you have a suggestion on how this text could be modified to make it clearer that we're defining a term here?